pkg/github/synchronizer.go

```diff
@@ -139,84 +140,56 @@ func (s *Synchronizer) Sync(ctx context.Context, teams []*v1alpha1.GitHubTeam) e
+			ListOptions: *listOpts,
 		}
-		for {
-			logins, resp, err := f(ctx, c, orgID, teamID, opt)
-			if err != nil {
-				return nil, fmt.Errorf("failed to get team member/invitation logins: %w", err)
-			}
 
-			res = sets.Union(res, logins)
-			if resp.NextPage == 0 {
-				break // No more pages to fetch
-			}
-			opt.Page = resp.NextPage
+		members, resp, err := c.Teams.ListTeamMembersByID(ctx, orgID, teamID, opts)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list team membership: %w", err)
 		}
-	}
-	return res, nil
-}
 
-func listActiveTeamMembers(ctx context.Context, c *github.Client, orgID, teamID int64, opt *github.ListOptions) ([]string, *github.Response, error) {
-	o := &github.TeamListTeamMembersOptions{
-		Role:        "all",
-		ListOptions: *opt,
-	}
-	members, resp, err := c.Teams.ListTeamMembersByID(ctx, orgID, teamID, o)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to list team membership: %w", err)
-	}
-	logins := make([]string, 0, len(members))
-	for _, m := range members {
-		// just checking, login should be provided for active members.
-		if m.GetLogin() != "" {
-			logins = append(logins, m.GetLogin())
+		for _, m := range members {
+			// just checking, login should be provided for active members.
+			if v := m.GetLogin(); v != "" {
+				loginsMap[v] = struct{}{}
+			}
 		}
+		return resp, nil
+	}); err != nil {
+		return nil, err
 	}
-	return logins, resp, nil
-}
 
-func listPendingTeamInvitations(ctx context.Context, c *github.Client, orgID, teamID int64, opt *github.ListOptions) ([]string, *github.Response, error) {
-	logger := logging.FromContext(ctx)
+	if err := Paginate(func(listOpts *github.ListOptions) (*github.Response, error) {
+		invitations, resp, err := c.Teams.ListPendingTeamInvitationsByID(ctx, orgID, teamID, listOpts)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list team invitations: %w", err)
+		}
 
-	invitations, resp, err := c.Teams.ListPendingTeamInvitationsByID(ctx, orgID, teamID, opt)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to list team invitations: %w", err)
-	}
-	logins := make([]string, 0, len(invitations))
-	for _, inv := range invitations {
-		// login could be missing if the invitation is sent to an email.
-		if inv.GetLogin() == "" {
-			logger.WarnContext(
-				ctx,
-				`skip checking invitation due to missing GitHub user login, please check
-the invitation manually to make sure it is valid`,
-				"invitation", inv,
-				"team_id", teamID,
-				"org_id", orgID,
-			)
-			continue
+		for _, inv := range invitations {
+			// login could be missing if the invitation is sent to an email.
+			if v := inv.GetLogin(); v != "" {
+				loginsMap[v] = struct{}{}
+			}
 		}
-		logins = append(logins, inv.GetLogin())
+		return resp, nil
+	}); err != nil {
+		return nil, err
 	}
-	return logins, resp, nil
+
+	// convert logins to a slice and sort
+	logins := make([]string, 0, len(loginsMap))
+	for k := range loginsMap {
+		logins = append(logins, k)
+	}
```

Comment:
Use https://pkg.go.dev/golang.org/x/exp/maps#Keys?

Reply:
Code [is identical](https://cs.opensource.google/go/x/exp/+/8a7402ab:maps/maps.go;l=10-16), but brings in unnecessary generics and isn't in the stdlib.